var/spack/repos/builtin/packages/charmpp/package.py

```diff
@@ -56,10 +58,18 @@ class Charmpp(Package):
         "backend",
         default="netlrts",
         values=("mpi", "multicore", "netlrts", "verbs", "gni",
-                "ofi", "pami", "pamilrts"),
+                "ucx", "ofi", "pami", "pamilrts"),
         description="Set the backend to use"
     )
 
+    # https://github.com/UIUC-PPL/charm/issues/2477
+    variant(
+        "ucx-pmi",
+        default="simplePMI",
```

Comment:
As far as I know, PMIx is the version with the best support in Charm++ currently, maybe this could be the default.

Reply:
For both ucx and non-ucx installs ?

Reply:
I think PMI is only used when building with ucx, but @nitbhat may be able to confirm.

Reply:
All of gni, ofi and ucx use PMI.